x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/authc/Subject.java

```diff
@@ -176,15 +176,42 @@ private boolean isEmptyRoleDescriptorsBytes(BytesReference roleDescriptorsBytes)
     // This following fixed role descriptor is for fleet-server BWC on and before 7.14.
     // It is fixed and must NOT be updated when the fleet-server service account updates.
     // Package private for testing
-    static final BytesArray FLEET_SERVER_ROLE_DESCRIPTOR_BYTES_V_7_14 = new BytesArray(
-        "{\"elastic/fleet-server\":{\"cluster\":[\"monitor\",\"manage_own_api_key\"],"
-            + "\"indices\":[{\"names\":[\"logs-*\",\"metrics-*\",\"traces-*\",\"synthetics-*\","
-            + "\".logs-endpoint.diagnostic.collection-*\"],"
-            + "\"privileges\":[\"write\",\"create_index\",\"auto_configure\"],\"allow_restricted_indices\":false},"
-            + "{\"names\":[\".fleet-*\"],\"privileges\":[\"read\",\"write\",\"monitor\",\"create_index\",\"auto_configure\"],"
-            + "\"allow_restricted_indices\":false}],\"applications\":[],\"run_as\":[],\"metadata\":{},"
-            + "\"transient_metadata\":{\"enabled\":true}}}"
-    );
+    static final BytesArray FLEET_SERVER_ROLE_DESCRIPTOR_BYTES_V_7_14 = new BytesArray("""
+        {
+          "elastic/fleet-server": {
+            "cluster": [ "monitor", "manage_own_api_key" ],
+            "indices": [
+              {
+                "names": [
+                  "logs-*",
+                  "metrics-*",
+                  "traces-*",
+                  "synthetics-*",
+                  ".logs-endpoint.diagnostic.collection-*"
+                ],
+                "privileges": [ "write", "create_index", "auto_configure" ],
+                "allow_restricted_indices": false
+              },
+              {
+                "names": [ ".fleet-*" ],
+                "privileges": [
+                  "read",
+                  "write",
+                  "monitor",
+                  "create_index",
+                  "auto_configure"
+                ],
+                "allow_restricted_indices": false
+              }
+            ],
+            "applications": [],
+            "run_as": [],
+            "metadata": {},
+            "transient_metadata": {
+              "enabled": true
+            }
+          }
+        }""");
```

Comment:
Do we want this formatting?
We push those bytes around the cluster and store them in ThreadContext, what's the impact of a 25% increase in the size of the string?

Reply:
Good question. I just thought it looks nicer in this format and didn't think of other implications. We don't store it in threadContext or sent it across nodes. But we do parse `RoleDescriptor` from it and more importantly computing message digest for it. So I think it's better not to have the format. 

One alternative to reverting the change is to add a call of `.replaceAll("[\\n ]", "")` at the end of the text block. So it can leverage the better looking format without paying most of the runtime cost. This approach is not very future proof. But since the value is meant to be immutable for this variable. It is not really a concern.

I am happy to just simply revert given the main purpose of this PR is to getting rid of unused variable. What do you think?

Reply:
I decided to revert the format change.